### PR TITLE
Fix migrations and add migration CI checks

### LINKED ISSUE #70: Add `try-runtime` CI job for checking the migrations
We should add some CI job that is running `try-runtime` to check if all migrations are correct.

CC @liamaharon maybe something that you could do?

### PR DESCRIPTION
**Pull request by liamaharon**
Closes #70 

The error with Asset Hub Kusama will be fixed when [this commit](https://github.com/paritytech/polkadot-sdk/commit/db3fd687262c68b115ab6724dfaa6a71d4a48a59) lands in the runtime.

### CODE REVIEW

File: .github/workflows/check-migrations.yml
Comment (bkchr) [inline .github/workflows/check-migrations.yml:70]: Will make this the job fail if we overrun the weight?

File: .github/workflows/check-migrations.yml
Comment (bkchr) [inline .github/workflows/check-migrations.yml:74]: We not all checks?

File: .github/workflows/check-migrations.yml
Comment (bkchr) [inline .github/workflows/check-migrations.yml:56]: We could also just build it our own here using cargo.

Comment (bkchr) [reply .github/workflows/check-migrations.yml:56]: `cargo build --profile production -p ${{ matrix.runtime.package }}`

That should it be.

**Review by bkchr — COMMENTED** [review]
Ty! Looking good, just some questions.

File: .github/workflows/check-migrations.yml
Comment (liamaharon) [reply .github/workflows/check-migrations.yml:70]: After this is merged, yes https://github.com/paritytech/try-runtime-cli/pull/46

File: .github/workflows/check-migrations.yml
Comment (liamaharon) [reply .github/workflows/check-migrations.yml:74]: The disabled checks here are try-state invariants. I still need to test those on all chains to make sure they pass. Don't want to turn it on yet just in case there're some jobs fail when in reality everything about the migrations is fine.

**Comment by liamaharon** [comment]
fyi the error with Asset Hub Kusama will be fixed when [this commit](https://github.com/paritytech/polkadot-sdk/commit/db3fd687262c68b115ab6724dfaa6a71d4a48a59) lands in the runtime.

**Review by bkchr — APPROVED** [review]

File: .github/workflows/check-migrations.yml
Comment (ggwpez) [inline .github/workflows/check-migrations.yml:43]:
```suggestion
          curl -sL https://github.com/paritytech/try-runtime-cli/releases/download/v0.4.0/try-runtime-x86_64-unknown-linux-musl -o try-runtime
```

File: .github/workflows/check-migrations.yml
Comment (ggwpez) [inline .github/workflows/check-migrations.yml:56]: Maybe its worth it to eventually build these with `srtool`, to ensure that the runtime we test is the exact same that we will deploy.  
But we can just start like this for sure.

File: .github/workflows/check-migrations.yml
Comment (ggwpez) [reply .github/workflows/check-migrations.yml:69]: You could also first create a snap and then once run with all checks and once with only pre-and-post. The first one would only be good for logging, not erroring.

**Review by ggwpez — APPROVED** [review]
Nice1

**Review by xlc — APPROVED** [review]

**Comment by liamaharon** [comment]
Please don't merge yet, need to fix syntax error

**Review by shawntabrizi — APPROVED** [review]

**Comment by liamaharon** [comment]
Thank you for the reviews :) 

I will fix the failing migrations, then this will be ready to merge.

File: .github/workflows/runtimes-matrix.json
Comment (ordian) [inline .github/workflows/runtimes-matrix.json:19]:
```suggestion
    "path": "system-parachains/gluttons/glutton-kusama",
    "is_relay": false
```

File: system-parachains/asset-hubs/asset-hub-kusama/src/lib.rs
Comment (bkchr) [inline system-parachains/asset-hubs/asset-hub-kusama/src/lib.rs:114]:
```suggestion
	spec_version: 1_000_000,
```

File: system-parachains/bridge-hubs/bridge-hub-kusama/src/lib.rs
Comment (bkchr) [inline system-parachains/bridge-hubs/bridge-hub-kusama/src/lib.rs:157]:
```suggestion
	spec_version: 1_000_000,
```

File: relay/polkadot/src/lib.rs
Comment (bkchr) [inline relay/polkadot/src/lib.rs:1601]:
```suggestion
		paras_registrar::migration::VersionCheckedMigrateToV1<Runtime, ParachainsToUnlock>,

		runtime_parachains::scheduler::migration::v1::MigrateToV1<Runtime>
```

File: relay/polkadot/src/lib.rs
Comment (bkchr) [inline relay/polkadot/src/lib.rs:134]:
```suggestion
	spec_version: 1_000_001,
```

File: relay/kusama/src/lib.rs
Comment (bkchr) [inline relay/kusama/src/lib.rs:141]:
```suggestion
	spec_version: 1_000_001,
```

**Review by bkchr — APPROVED** [review]

**Pull request merged**
